### PR TITLE
feat(hapi): automate new election terms

### LINKED ISSUE #118: Automate new election terms
New elections need to trigger an event that brings in election results and creates a new election term

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- Resolve #118 

### Steps to test

1.  See changes in the hapi folder

#### CheckList

- [X] Follow proper Markdown format
- [X] The content is adequate
- [X] The content is available in both english and spanish
- [X] I Ran a spell check

### CODE REVIEW

File: hapi/src/constants/serviceMessage.constant.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const MESSAGES = {
   delegates: 'UPDATE ACTUAL EDEN DELEGATES AND ELECTIONS',
-  historicDelegates: 'UPDATE HISCTORIC EDEN DELEGATES AND HISTORIC ELECTIONS'
+  historicDelegates: 'UPDATE HISCTORIC EDEN DELEGATES AND HISTORIC ELECTIONS',
+  uncleimedFunds: 'UPDATE UNCLEIMED FUNDS'
```
Comment (xavier506) [inline hapi/src/constants/serviceMessage.constant.js:4]: UNCLAIMED

File: .github/workflows/push-prod-environment.yaml [hunk 2]
```diff
@@ -75,6 +75,11 @@ jobs:
           HAPI_RATE_HISTORY: ${{secrets.HAPI_RATE_HISTORY }}
           HAPI_HYPERION_START_AT: '2022-07-01T00:00:00.000+00:00'
           HAPI_EDEN_ELECTION_INTERVAL: 7889400
+          HAPI_EDEN_HISTORIC_ELECTION: "[
```
Comment (leisterfrancisco) [inline .github/workflows/push-prod-environment.yaml:78]: Load this variable from secrets.

It is not a secret value but it is easier to update only a variable than having to update the code en redeploying

File: .vscode/settings.json [hunk 3]
```diff
@@ -7,5 +7,6 @@
     "**/.yarn": true,
     "**/.pnp.*": true
   },
-  "prettier.prettierPath": "./webapp/.yarn/sdks/prettier/index.js"
+  "prettier.prettierPath": "./webapp/.yarn/sdks/prettier/index.js",
+  "eslint.workingDirectories": [{ "mode": "auto" }]
```
Comment (leisterfrancisco) [inline .vscode/settings.json:11]: what is it for?

File: hapi/src/utils/services.util.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+const nextElectionDate = async () => {
+  try {
+    const response = await loadTableData({ next_key: null }, 'elect.curr')
+    const nextElection = response.rows[0]
+    const now = moment(new Date())
+    const end = moment(nextElection[1].start_time).add(1, 'day')
+    const duration = moment.duration(end.diff(now)).asSeconds()
+
+    return duration > 0 ? duration : 3600
```
Comment (leisterfrancisco) [inline hapi/src/utils/services.util.js:16]: what does `3600` mean, put this value in the constants folder

File: hapi/src/utils/services.util.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    const now = moment(new Date())
+    const end = moment(nextDistribution[1].distribution_time).add(1, 'day')
+    const duration = moment.duration(end.diff(now)).asSeconds()
+
+    return duration > 0 ? duration : 3600
+  } catch (error) {
+    console.log(error)
+
+    return 3600
```
Comment (leisterfrancisco) [inline hapi/src/utils/services.util.js:38]: same here

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 3 above)
Comment (warnerHurtado) [reply .vscode/settings.json:11]: That is a Teto's recomendation to not see an error in imports

**Review by leisterfrancisco — APPROVED** [review]

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**